server/imports/core/connection/index.js

```diff
@@ -1,77 +1,154 @@
+      parsedUrl.server_options.socketOptions.socketTimeoutMS
+        ? parsedUrl.server_options.socketOptions.socketTimeoutMS
+        : '';
     connection.ssl.enabled = !!parsedUrl.server_options.ssl;
   }
 
-  connection.options.replicaSetName = (parsedUrl.rs_options && parsedUrl.rs_options.rs_name) ? parsedUrl.rs_options.rs_name : '';
+  connection.options.replicaSetName =
+    parsedUrl.rs_options && parsedUrl.rs_options.rs_name
+      ? parsedUrl.rs_options.rs_name
+      : '';
   connection.options.readPreference = parsedUrl.db_options.read_preference;
 };
 
 const setAuthToConnectionFromParsedUrl = function (connection, parsedUrl) {
-  connection.authenticationType = parsedUrl.db_options.authMechanism ? parsedUrl.db_options.authMechanism.toLowerCase().replace(new RegExp('-', 'g'), '_') : '';
-  if (connection.authenticationType) connection[connection.authenticationType] = {};
-  if (parsedUrl.db_options.gssapiServiceName && connection.authenticationType === 'gssapi') connection.gssapi.serviceName = parsedUrl.db_options.gssapiServiceName;
+  connection.authenticationType = parsedUrl.db_options.authMechanism
+    ? parsedUrl.db_options.authMechanism
+        .toLowerCase()
+        .replace(new RegExp('-', 'g'), '_')
+    : '';
+  if (connection.authenticationType) {
+    connection[connection.authenticationType] = {};
+  }
+  if (
+    parsedUrl.db_options.gssapiServiceName &&
+    connection.authenticationType === 'gssapi'
+  ) {
+    connection.gssapi.serviceName = parsedUrl.db_options.gssapiServiceName;
+  }
   if (connection.authenticationType === 'mongodb_x509') delete connection.ssl;
 
   if (parsedUrl.auth) {
     // if auth exists there should be an authentication, even there's no authMechanism set
-    connection.authenticationType = connection.authenticationType || 'scram_sha_1';
-    connection[connection.authenticationType] = connection[connection.authenticationType] || {};
-    connection[connection.authenticationType].username = parsedUrl.auth.user ? parsedUrl.auth.user : '';
-    connection[connection.authenticationType].password = parsedUrl.auth.password ? parsedUrl.auth.password : '';
+    connection.authenticationType =
+      connection.authenticationType || 'scram_sha_1';
+    connection[connection.authenticationType] =
+      connection[connection.authenticationType] || {};
+    connection[connection.authenticationType].username = parsedUrl.auth.user
+      ? parsedUrl.auth.user
+      : '';
+    connection[connection.authenticationType].password = parsedUrl.auth.password
+      ? parsedUrl.auth.password
+      : '';
   }
-  if (connection.authenticationType === 'mongodb_cr' || connection.authenticationType === 'scram_sha_1' || connection.authenticationType === 'scram_sha_256') {
-    connection[connection.authenticationType].authSource = parsedUrl.db_options.authSource ? parsedUrl.db_options.authSource : connection.databaseName;
+  if (
+    connection.authenticationType === 'mongodb_cr' ||
```

Comment:
Codacy has a fix for the issue: ['||' should be placed at the beginning of the line.](https://app.codacy.com/gh/nosqlclient/nosqlclient/pullRequest?prid=7730964)
```suggestion
    connection.authenticationType === 'mongodb_cr'
```